To: vim_dev@googlegroups.com
Subject: Patch 7.3.095
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.095
Problem:    Win32: In Chinese tear-off menu doesn't work. (Weasley)
Solution:   Use menu_name_equal().  (Alex Jakushev)
Files:	    src/menu.c


*** ../vim-7.3.094/src/menu.c	2010-08-15 21:57:25.000000000 +0200
--- src/menu.c	2011-01-04 17:41:38.000000000 +0100
***************
*** 1512,1519 ****
  {
  #ifdef FEAT_MULTI_LANG
      if (menu->en_name != NULL
! 	    && (menu_namecmp(name,menu->en_name)
! 		|| menu_namecmp(name,menu->en_dname)))
  	return TRUE;
  #endif
      return menu_namecmp(name, menu->name) || menu_namecmp(name, menu->dname);
--- 1512,1519 ----
  {
  #ifdef FEAT_MULTI_LANG
      if (menu->en_name != NULL
! 	    && (menu_namecmp(name, menu->en_name)
! 		|| menu_namecmp(name, menu->en_dname)))
  	return TRUE;
  #endif
      return menu_namecmp(name, menu->name) || menu_namecmp(name, menu->dname);
***************
*** 2342,2348 ****
  
  	while (menu != NULL)
  	{
! 	    if (STRCMP(name, menu->name) == 0 || STRCMP(name, menu->dname) == 0)
  	    {
  		if (menu->children == NULL)
  		{
--- 2342,2348 ----
  
  	while (menu != NULL)
  	{
! 	    if (menu_name_equal(name, menu))
  	    {
  		if (menu->children == NULL)
  		{
*** ../vim-7.3.094/src/version.c	2010-12-30 14:57:03.000000000 +0100
--- src/version.c	2011-01-04 17:43:41.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     95,
  /**/

-- 
If your nose runs, and your feet smell, you might be upside down.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
